packages/expo-modules-core/android/src/main/cpp/JavaScriptModuleObject.cpp

```diff
@@ -0,0 +1,137 @@
+}
+
+jni::local_ref<react::ReadableNativeArray::javaobject>
+JavaScriptModuleObject::callSyncMethod(jni::local_ref<jstring> &&name,
+                                       react::ReadableNativeArray::javaobject &&args) {
+  static const auto method = JavaScriptModuleObject::javaClassLocal()
+    ->getMethod<react::ReadableNativeArray::javaobject(
+      jni::local_ref<jstring>,
+      react::ReadableNativeArray::javaobject)>(
+      "callSyncMethod"
+    );
+
+  return method(javaPart_.get(), std::move(name), args);
+}
+
+JavaScriptModuleObject::HostObject::HostObject(
+  JavaScriptModuleObject *jsModule) : jsModule(jsModule) {}
+
+jsi::Value JavaScriptModuleObject::HostObject::get(jsi::Runtime &runtime,
+                                                   const jsi::PropNameID &name) {
+  auto cName = name.utf8(runtime);
+  auto metadataRecord = jsModule->methodsMetadata.find(cName);
+  if (metadataRecord == jsModule->methodsMetadata.end()) {
+    return jsi::Value::undefined();
+  }
+  auto metadata = metadataRecord->second;
+
+  if (metadata.body == nullptr) {
+    auto body = jsi::Function::createFromHostFunction(
+      runtime,
+      jsi::PropNameID::forAscii(runtime, metadata.name),
+      metadata.args,
+      [&jsModule = jsModule, cName](
+        jsi::Runtime &rt,
+        const jsi::Value &thisValue,
+        const jsi::Value *args,
+        size_t count
+      ) -> jsi::Value {
+        auto dynamicArray = folly::dynamic::array();
+        for (int i = 0; i < count; i++) {
+          auto &arg = args[i];
+          dynamicArray.push_back(jsi::dynamicFromValue(rt, arg));
+        }
+
+        auto result = jsModule->callSyncMethod(
+          jni::make_jstring(cName),
+          react::ReadableNativeArray::newObjectCxxArgs(std::move(dynamicArray)).get()
+        );
+
+        if (result == nullptr) {
+          return jsi::Value::undefined();
+        }
+
+        return jsi::valueFromDynamic(rt, result->cthis()->consume())
+          .asObject(rt)
+          .asArray(rt)
+          .getValueAtIndex(rt, 0);
+      }
+    );
+
+    metadata.body = std::make_shared<jsi::Function>(std::move(body));
+  }
+
+  return jsi::Value(runtime, *metadata.body);
+}
+
+void
+JavaScriptModuleObject::HostObject::set(jsi::Runtime &runtime, const jsi::PropNameID &name,
+                                        const jsi::Value &value) {
+  std::string message("RuntimeError: Cannot override the host object for expo module '");
+  message += name.utf8(runtime);
+  message += "'.";
+  throw jsi::JSError(runtime, message);
+}
+
+std::vector<jsi::PropNameID>
+JavaScriptModuleObject::HostObject::getPropertyNames(jsi::Runtime &rt) {
+  auto metadata = jsModule->methodsMetadata;
+  std::vector<jsi::PropNameID> result;
+  std::transform(
+    metadata.begin(),
+    metadata.end(),
+    std::back_inserter(result),
+    [&rt](const auto &kv) {
+      return jsi::PropNameID::forUtf8(rt, kv.first);
+    }
+  );
+
+  return result;
+}
+}
```

Comment:
I think it's a good practice to add such comments after the closing bracket

```suggestion
} // namespace expo
```